### PR TITLE
refactor(rxStatusColumn): EO-1142 Element Table Bulk Status Column

### PR DESCRIPTION
**Pull request by manojdas2212**
JIRA: https://jira.rax.io/browse/EO-1142

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] Design LGTM
- [x] QE LGTM

Before
------
screen shot 2016-09-02 at 2 10 52 pm

After
------
screen shot 2016-09-02 at 2 11 15 pm
screen shot 2016-09-02 at 2 11 24 pm

API Doc
------
screen shot 2016-09-02 at 2 11 42 pm

Search for "All Modules"
------
Search for rxStatusColumn
------
screen shot 2016-09-02 at 2 13 38 pm

Search for status
------
screen shot 2016-09-02 at 2 13 52 pm

Search for column
------
screen shot 2016-09-02 at 2 14 11 pm

Search for table
------
screen shot 2016-09-02 at 2 14 23 pm

### CODE REVIEW

**Comment by KalpeshBhandari** [comment]
@manojdas2212 Within table.statusColumn.html file two space indentation is required.

File: src/elements/Tables/docs/tables.docs.html [hunk 1]
```diff
@@ -1,3 +1,14 @@
+  <h2><rx-permalink>Status Column</rx-permalink>
```
Comment (KalpeshBhandari) [inline src/elements/Tables/docs/tables.docs.html]: @manojdas2212  These are at the top of the page and hence no indentation here.

**Comment by CITguy** [comment]
Please correct indentation in `src/elements/Tables/docs/examples/table.statusColumn.html`.

**Comment by CITguy** [comment]
In `src/elements/Tables/docs/examples/table.statusColumn.html`:

1. Please move the `ng-controller` attribute from the `<div>` on line 1 to the `<table>` on line 2.
2. Please remove the `<div>`

**Comment by CITguy** [comment]
@Droogans: what would you say about removing sorting tests for the rxStatusColumn examples? They're complicating the example, and they aren't visibly sorting as expected, anyway.

rxsortablecolumn-broken-sorting

Sorting is happening on the _data value_, not the translated status. I'd expect all of the ERROR statuses to be grouped together, PENDING statuses grouped together, etc. Since this isn't sorting as expected, I'd rather not have sorting at all.

**Comment by CITguy** [comment]
@ElementE, and @ssalinas24: ^^ what say you about the sorting?

**Comment by Droogans** [comment]
@CITguy I can't really say anything about those tests...when the component was first created, the sorting was determined to be based on the name (whereas [I submitted an idea](https://github.com/rackerlabs/encore-ui/pull/735#discussion-diff-22966801) to make it based on a combination of color + status icon), it wasn't accepted.

The test asserts that it sorts "correctly" though so I'm hesitant to remove it. Is it providing a false sense of security? I'm not sure what the problem is with keeping the tests around.

**Comment by CITguy** [comment]
@Droogans: Functionally, there's no problem, but it's over complicating the example to show how to implement a status column. Sorting isn't necessary to achieve that.

**Comment by Droogans** [comment]
What about wrapping the table in `rx-debug`, and then making a copy of it for the demo that disabled sorting? The way it handles sort logic is exposed, and should be tested. But probably not shown to users.

**Comment by CITguy** [comment]
I'd be open to that.

**Comment by ssalinas24** [comment]
@CITguy and @ElementE In regards to the  Status sorting, it seems to be alphabetized correctly to me. I would assume if more Active statuses existed then they would grouped together at the top and so on for other statuses.

**Comment by CITguy** [comment]
I'm going to push up the change to separate into two tables. The rxExample demo shouldn't have sorting, so I'll make a second table that includes sorting within an rxDebug.

**Comment by manojdas2212** [comment]
@KalpeshBhandari @CITguy Updated changes please review

File: src/elements/Tables/docs/examples/table.statusColumn.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          Status
+        </rx-sortable-column>
+      </th>
+      <th class="column-title">
+        Title
+      </th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr ng-repeat="server in servers | orderBy: sort.predicate:sort.reverse ">
```
Comment (sundip) [inline src/elements/Tables/docs/examples/table.statusColumn.html]: Please remove the space before the final quote.

File: src/elements/Tables/docs/examples/table.statusColumn.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      <!-- Both `api` and `tooltip-content` are optional -->
+      <td rx-status-column status="{{ server.status }}" api="{{ server.api }}" tooltip-content="{{ server.status }}"></td>
```
Comment (sundip) [inline src/elements/Tables/docs/examples/table.statusColumn.html]: The character count of this line is 122.

**Comment by manojdas2212** [comment]
@sundip Updated the changes please review

**Comment by sundip** [comment]
Dev: LGTM

**Comment by CITguy** [comment]
Dev LGTM

**Comment by CITguy** [comment]
@ElementE or @ssalinas24: can either of you provide a Design LGTM, please?

**Comment by ElementE** [comment]
Design LGTM

**Pull request merged**